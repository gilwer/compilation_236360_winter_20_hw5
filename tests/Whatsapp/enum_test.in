enum days {SUNDAY, MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY};
enum months {JANUARY, FEBRUARY, MARCH, APRIL, MAY, JUNE, JULY, AUGUST, SEPTEMBER, OCTOBER, NOVEMBER, DECEMBER};

int getDayNum(enum days d) {
    return (int)d;
}

bool beforeApril(enum months m) {
    enum numbers {ONE, TWO, THREE, FOUR};
    return true and true and (int)m < (int)FOUR and true;
}

void isEqual(enum months m, byte n) {
    printi((int)m);
    if ((int)m == n)
        print("... is equal to ...");
    else
        print("... is different from ...");
    printi(n);
}

void main() {
    enum days x = WEDNESDAY;
    enum days y = FRIDAY;
    if ((int)x == (int)y)
        print("time has collapsed");
    else
        print("Wednesday is not Friday");

    enum months z = JANUARY;
    if ((int)x != (int)z) {
        printi((int)x);
    }

    printi(getDayNum(FRIDAY));

    bool isBeforeApril = beforeApril(MARCH);
    if (isBeforeApril)
        print("March is before April");
    isBeforeApril = beforeApril(MAY);
    if (isBeforeApril)
        print("May is before April");

    enum months zz = z;
    isEqual(zz, 2b);
}
