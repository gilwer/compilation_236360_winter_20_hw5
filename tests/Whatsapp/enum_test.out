Wednesday is not Friday
3
5
March is before April
0
... is different from ...
2
